server/api/event/index.put.ts

```diff
@@ -0,0 +1,96 @@
+// PUT /api/events
+// Endpoint for overwriting existing events
+
+export default defineEventHandler(async (event) => {
+  const { user } = event.context
+
+  const {
+    companyUID,
+    eventUID,
+    capacity,
+    date,
+    description,
+    limitedCapacity,
+    location,
+    title,
+    eventType,
+  } = await readBody(event)
+
+  // Check if user is authorized
+  if (hasAccess(user, ['company'])) {
+    if (user.companyUID !== companyUID)
+      throw createError({
+        statusCode: 401,
+        statusMessage: 'Company users cannot update events for other companies',
+      })
+  } else if (!hasAccess(user, ['admin']))
+    throw createError({
+      statusCode: 401,
+      statusMessage: 'User not authenticated',
+    })
+
+  // check if data is defined.
+  if (

```

Comment:
This makes it so all values are still required when making the request. Maybe change this to check that values are not null, so that non of the fields in the database can be deleted, only changed.